components/layout_2020/lib.rs

```diff
@@ -29,56 +29,57 @@ use app_units::Au;
 pub use flow::BoxTree;
 pub use fragment_tree::FragmentTree;
 use geom::AuOrAuto;
+use style::logical_geometry::WritingMode;
 use style::properties::ComputedValues;
 
 use crate::geom::LogicalVec2;
 
 /// A containing block useful for calculating inline content sizes, which may
 /// have inline sizes that depend on block sizes due to aspect ratio.
-pub(crate) struct IndefiniteContainingBlock<'a> {
+pub(crate) struct IndefiniteContainingBlock {
     pub size: LogicalVec2<AuOrAuto>,
-    pub style: &'a ComputedValues,
+    pub writing_mode: WritingMode,
 }
 
-impl<'a> IndefiniteContainingBlock<'a> {
-    fn new_for_style(style: &'a ComputedValues) -> Self {
-        Self::new_for_style_and_block_size(style, AuOrAuto::Auto)
+impl IndefiniteContainingBlock {
+    fn new_for_wm(writing_mode: WritingMode) -> Self {
```

Comment:
I think we should avoid the 'wm' abbreviation because that usually means "window manager."

```suggestion
    fn new_for_writing_mode(writing_mode: WritingMode) -> Self {
```